python/src/viewer.py

```diff
@@ -516,33 +512,79 @@ def __init__(
+                        "dotdash": "-.",
+                        "longdash": "--",
+                        "twodash": "--"
+                    }
                     try:
-                        contour_kw["linestyles"] = lineStyleDict[
+                        contour_kw["linestyles"] = contourLineStyleDict[
                             drawable.getLineStyle()
                         ]
                     except KeyError:
                         warnings.warn("-- Unknown line style")
-                if "colors" not in contour_kw_default:
+                if not contour.isFilled() and ("linewidths" not in contour_kw_default) and (
+                    "lw" not in contour_kw_default
+                ):
+                    contour_kw["linewidths"] = drawable.getLineWidth()
+                if "cmap" not in contour_kw_default and contour.getColorMap():
+                    contour_kw["cmap"] = contour.getColorMap()
+                if "colors" not in contour_kw_default and "cmap" not in contour_kw:
                     contour_kw["colors"] = [drawable.getColorCode()]
-                contourset = self._ax[0].contour(X, Y, Z, **contour_kw)
-                if drawable.getDrawLabels():
+                if "alpha" not in contour_kw_default:
+                    contour_kw["alpha"] = contour.getAlpha()
+                if "vmin" not in contour_kw_default and contour.isVminUsed():
+                    contour_kw["vmin"] = contour.getVmin()
+                if "vmax" not in contour_kw_default and contour.isVmaxUsed():
+                    contour_kw["vmax"] = contour.getVmax()
+                if "norm" not in contour_kw_default:
+                    if not hasattr(cls, 'AsinhNorm'):
+                        # matplotlib before 3.6 does not support norms as strings
+                        try:
+                            normDict = {
+                                'linear': cls.Normalize(),
+                                'log': cls.LogNorm(),
+                                'symlog': cls.SymLogNorm(linthresh=0.03) if matplotlib.__version__ < "3.2.0"
+                                else cls.SymLogNorm(linthresh=0.03, base=10)
+                            }
+                            contour_kw["norm"] = normDict[contour.getNorm()]
+                        except KeyError:
+                            warnings.warn("-- Unknown norm " + contour.getNorm())
+                    else:
+                        contour_kw["norm"] = contour.getNorm()
+                if "extend" not in contour_kw_default:
+                    contour_kw["extend"] = contour.getExtend()
+                if contour.isFilled() and "hatches" not in contour_kw_default and contour.getHatches():
+                    contour_kw["hatches"] = [hatch for hatch in contour.getHatches()]
+                contourset = self._ax[0].contour(X, Y, Z, **contour_kw) if not contour.isFilled()\
+                    else self._ax[0].contourf(X, Y, Z, **contour_kw)
```

Comment:
It seems to me that this code is quite difficult to read with this long ternary operator. It would be simpler with a "if/else".